subprojects/language-java/src/integTest/groovy/org/gradle/java/compile/incremental/SourceIncrementalJavaCompilationIntegrationTest.groovy

```diff
@@ -471,7 +473,7 @@ compileTestJava.options.incremental = true
 
         then:
         outputs.recompiledClasses("A", "B", "C")
-        output.contains("Cannot infer source root(s) for input with type `ArrayList`. Supported types are `File`, `DirectoryTree` and `SourceDirectorySet`.")
+        output.contains("Cannot infer source root(s) for source `file '${textFile.absolutePath}'`. Only things resolving to directory trees are supported.")
```

Comment:
I'm not sure if this message is useful now to the user. Can we explicitly call out type names as we used to, even if just as examples?

Reply:
I am calling out types again.